M66/M66ATParser/M66ATParser.cpp

```diff
@@ -258,7 +258,9 @@ bool M66ATParser::close(int id)
     //May take a second try if device is busy
     for (unsigned i = 0; i < 2; i++) {
         if (tx("AT+QICLOSE=%d", id)
-            && scan("%d, CLOSE OK", &id_resp)) {
+            && scan("%d, CLOSE OK", &id_resp)
+            && (id == id_resp)) {
+
             return true;
```

Comment:
If we try to close the wrong connection or if the wrong connection is closed there is a bigger problem anyway, do the simple version and just return if we closed the right connection.